src/power_grid_model/utils.py

```diff
@@ -190,7 +191,7 @@ def import_json_data(json_file: Path, data_type: str, *args, **kwargs) -> Datase
     if kwargs:
         warnings.warn(f"Provided keyword arguments {list(kwargs.keys())} are deprecated.", DeprecationWarning)
 
-    return _compatibility_deprecated_import_json_data(json_file=json_file, data_type=data_type)
+    return _compatibility_deprecated_import_json_data(json_file=json_file, data_type=data_type)  # type: ignore
```

Comment:
The data_type argument can be of `DataType` and this suppresion might not be needed

Reply:
The suppression comes because the function is deprecated, hence I am not updating it with the new feature, but just ignoring the possible type error. Nevertheless, it is still backwards compatible.